lib/src/importer/node_package.dart

```diff
@@ -0,0 +1,394 @@
+// Copyright 2023 Google Inc. Use of this source code is governed by an
+// MIT-style license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:collection/collection.dart';
+import 'package:sass/src/util/map.dart';
+import 'package:sass/src/util/nullable.dart';
+
+import '../importer.dart';
+import './utils.dart';
+import 'dart:convert';
+import '../io.dart';
+import 'package:path/path.dart' as p;
+
+/// An [Importer] that resolves `pkg:` URLs using the Node resolution algorithm.
+class NodePackageImporterInternal extends Importer {
+  final Uri entryPointURL;
+
+  /// Creates a Node Package Importer with the associated entry point url
+  NodePackageImporterInternal(this.entryPointURL);
```

Comment:
I'd rather keep this parameter as a path, but if it has to be a URL I think we should probably consume this as a path and only convert it to a URL when we set the internal field. As a rule of thumb, when something is always going to represent a path on disk, it's better to keep it as a path until the last moment.